cmd/snap-fde-keymgr/main.go

```diff
@@ -86,11 +105,29 @@ func (c *cmdAddRecoveryKey) Execute(args []string) error {
 }
 
 func (c *cmdRemoveRecoveryKey) Execute(args []string) error {
-	if err := keymgrRemoveRecoveryKeyFromLUKSDevice(c.Device); err != nil {
-		return fmt.Errorf("cannot remove recovery key from LUKS device: %v", err)
+	if len(c.Authorization) != len(c.Device) {
+		return fmt.Errorf("cannot remove recovery key: mismatch in the number of devices and authorizations")
+	}
+	for i, dev := range c.Device {
+		authz := c.Authorization[i]
+		if authz == "keyring" {
+			if err := keymgrRemoveRecoveryKeyFromLUKSDevice(dev); err != nil {
+				return fmt.Errorf("cannot remove recovery key from LUKS device: %v", err)
+			}
+		} else if strings.HasPrefix(authz, "file:") {
+			authzKey, err := ioutil.ReadFile(authz[len("file:"):])
+			if err != nil {
+				return fmt.Errorf("cannot load authorization key: %v", err)
+			}
+			if err := keymgrRemoveRecoveryKeyFromLUKSDeviceUsingKey(authzKey, dev); err != nil {
+				return fmt.Errorf("cannot remove recovery key from device using authorization key: %v", err)
+			}
+		}
 	}
-	if err := os.Remove(c.KeyFile); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("cannot remove recovery key file: %v", err)
+	for _, kf := range c.KeyFile {
+		if err := os.Remove(kf); err != nil && !os.IsNotExist(err) {
```

Comment:
Should this maybe first collect all errors and try to delete as many keyfiles as possible before returning?